lib/postgres_v3_experimental.dart

```diff
@@ -0,0 +1,267 @@
+  factory PgResult(
+          List<PgResultRow> rows, int affectedRows, PgResultSchema schema) =
+      _PgResult;
+}
+
+class _PgResult extends DelegatingList<PgResultRow> implements PgResult {
+  @override
+  final int affectedRows;
+
+  @override
+  final PgResultSchema schema;
+
+  final List<PgResultRow> rows;
+
+  _PgResult(this.rows, this.affectedRows, this.schema) : super(rows);
+
+  @override
+  String toString() {
+    return 'PgResult(schema = $schema, affectedRows = $affectedRows, rows = $rows)';
+  }
+}
+
+abstract class PgResultRow implements List<Object?> {
+  PgResultSchema get schema;
+}
+
+class PgResultSchema {
+  final List<PgResultColumn> columns;
+
+  PgResultSchema(this.columns);
+
+  @override
+  String toString() {
+    return 'PgResultSchema(${columns.join(', ')})';
+  }
+}
+
+class PgResultColumn {
+  final PgDataType type;
+  final String? tableName;
+  final int? tableOid;
+  final String? columnName;
+  final int? columnOid;
+
+  PgResultColumn({
+    required this.type,
+    this.tableName,
+    this.tableOid,
+    this.columnName,
+    this.columnOid,
+  });
+
+  @override
+  String toString() {
+    final buffer = StringBuffer('${type.name} ');
+    if (tableName != null && tableName != '') {
+      buffer
+        ..write(tableName)
+        ..write('.');
+    } else if (tableOid != null && tableOid != 0) {
+      buffer
+        ..write('@$tableOid')
+        ..write('.');
+    }
+
+    if (columnName != null && columnName != '') {
+      buffer.write(columnName);
+    } else if (columnOid != null && columnOid != 0) {
+      buffer.write('@$columnOid');
+    }
+
+    return buffer.toString();
+  }
+}
+
+abstract class PgChannels {
+  Stream<String> operator [](String channel);
+  Future<void> notify(String channel, [String? payload]);
+  Future<void> cancelAll();
+}
+
+abstract class PgNotification {}
+
+class PgEndpoint {
+  final String host;
+  final int port;
+  final String database;
+  final String? username;
+  final String? password;
+  final bool requireSsl;
+  final bool isUnixSocket;
+
+  /// An optional [StreamChannelTransformer] sitting behind the postgres client
+  /// as implemented in the `posgres` package and the database server.
+  ///
+  /// The stream channel transformer is able to view, alter, drop, or inject
+  /// messages in either direction. This powerful tool can be used to implement
+  /// additional or custom functionality, but should also be used with caution
+  /// as altering the message flow might break internal invariants of this
+  /// package.
+  ///
+  /// For an example, see `example/v3/transformer.dart`.
+  final StreamChannelTransformer<BaseMessage, BaseMessage>? transformer;
+
+  PgEndpoint({
+    required this.host,
+    this.port = 5432,
+    required this.database,
+    this.username,
+    this.password,
+    this.requireSsl = false,
+    this.isUnixSocket = false,
+    this.transformer,
+  });
+
+  Future<PgConnection> connect({PgSessionSettings? sessionSettings}) {
```

Comment:
Is there any benefit of having such helper methods? We may end up duplicating and passing all the parameters everywhere...

Reply:
I'd hope we don't end up having too many parameters for that method :D But I see your point, I will remove this helper method.